### PR TITLE
10734 custom question hawaii

### PR DESCRIPTION
**Pull request by mchelen-gov**
## Description
https://github.com/department-of-veterans-affairs/va.gov-team/issues/10734

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/coronavirus-screener/containers/App.jsx [hunk 1]
```diff
@@ -3,26 +3,30 @@ import MetaTags from 'react-meta-tags';
...
+export default class App extends React.Component {
+  render() {
+    return (
+      <div className="covid-screener">
+        <MetaTags>
+          <meta name="robots" content="noindex" />
+        </MetaTags>
+        <div className="vads-l-grid-container">
+          <h1>COVID-19 screening tool</h1>
+          <h2>Site ID: {this.props.params.id}</h2>
```
Comment (batemapf) [inline src/applications/coronavirus-screener/containers/App.jsx:16]: Not sure if this is just scaffolding, but definitely want to check w/ design + content on adding in this text.

File: src/applications/coronavirus-screener/routes.jsx [hunk 2]
```diff
@@ -2,6 +2,6 @@ import React from 'react';
 import { Route } from 'react-router';
 import App from './containers/App.jsx';
 
-const routes = <Route path="/*" component={App} />;
+const routes = <Route path="/:id" component={App} />;
```
Comment (batemapf) [inline src/applications/coronavirus-screener/routes.jsx:5]: Confirming that the naked route (/covid19screen/) will not be impacted?

File: src/applications/coronavirus-screener/containers/App.jsx (hunk 1 above)
Comment (mchelen-gov) [reply src/applications/coronavirus-screener/containers/App.jsx]: oh yeah that was just a debugging placeholder

File: src/applications/coronavirus-screener/containers/App.jsx [hunk 3]
```diff
@@ -3,26 +3,30 @@ import MetaTags from 'react-meta-tags';
...
-      </MetaTags>
-      <div className="vads-l-grid-container">
-        <h1>COVID-19 screening tool</h1>
-        <div className="va-introtext">
-          <p>
-            Please answer the questions listed below. Share your results with
-            the staff member at the facility entrance.
-          </p>
-          <p>We won't store or share your data.</p>
+// eslint-disable-next-line react/prefer-stateless-function
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/coronavirus-screener/containers/App.jsx:6]: Is there a reason that `<App>` needs to be a class based component?

[the remark this answers is not in the document]
Comment (mchelen-gov) [reply src/applications/coronavirus-screener/containers/App.jsx:6]: @bkjohnson good question! i originally tried with `useParams()` https://reacttraining.com/react-router/web/api/Hooks/useparams but kept getting an error `Uncaught TypeError: Cannot read property 'match' of undefined` which seems to be caused by https://stackoverflow.com/a/59153982 but i wasn't sure how to fix it.


Here's a branch with that approach if you have any ideas: https://github.com/department-of-veterans-affairs/vets-website/tree/10734-custom-question-hawaii-func-comp

File: src/applications/coronavirus-screener/routes.jsx (hunk 2 above)
Comment (mchelen-gov) [reply src/applications/coronavirus-screener/routes.jsx]: it does if https://github.com/department-of-veterans-affairs/vets-website/blob/8ae0809076efa46f2de3f9d81bbae33a466fb9f7/src/applications/coronavirus-screener/tests/00.coronavirus-screener.e2e.spec.js passes! (btw nightwatch e2e tests run only in jenkins now)

**Review by Yuffster-usds — APPROVED** [review]
lgtm

File: src/applications/coronavirus-screener/containers/App.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/coronavirus-screener/containers/App.jsx:6]: What about something like:

```diff
diff --git a/src/applications/coronavirus-screener/containers/App.jsx b/src/applications/coronavirus-screener/containers/App.jsx
index e6d5385f88..f40f2212fb 100644
--- a/src/applications/coronavirus-screener/containers/App.jsx
+++ b/src/applications/coronavirus-screener/containers/App.jsx
@@ -3,7 +3,7 @@ import MetaTags from 'react-meta-tags';
 import MultiQuestionForm from '../components/MultiQuestionForm';
 import { questions, defaultOptions } from '../config/questions';
 
-export default function App() {
+export default function App({ params }) {
   return (
     <div className="covid-screener">
       <MetaTags>
@@ -21,6 +21,7 @@ export default function App() {
         <MultiQuestionForm
           questions={questions}
           defaultOptions={defaultOptions}
+          customId={params.id}
         />
       </div>
     </div>
```
  
If `params` is just a prop then it should get passed in as a value in the `props` argument of a functional component which can then be destructured.

[the remark this answers is not in the document]
Comment (mchelen-gov) [reply src/applications/coronavirus-screener/containers/App.jsx]: i think that did the trick, thank you!

**Comment by stephenBarrs** [comment]
@mchelen-gov running the review instance and manually testing `/covid19screen/459`, when selecting `Yes` to `Are you an employee or contractor` question, I am not seeing the Hawaii specific question come up and the app does not display the complete screen.  Have not had time to trace through the code yet.

**Comment by mchelen-gov** [comment]
@stephenBarrs interesting, good catch! selecting "yes" to first question and "no" to the rest does give me the result screen, but it does *not* show the travel question. i added some e2e tests which should now fail, and i'll look into why this is happening

**Review by stephenBarrs — APPROVED** [review]
LGTM.  I cannot run the review instance but everything else looks good.

**Pull request merged**